Lesson2

YarRus12 opened the pull request:
Направляю PR для сдачи второго домашнего задания

YarRus12 committed:
Выполнена часть 1 задания, но стоит перепроверить с конспектом лекции

YarRus12 committed:
Подготовлены заготовки для 2-3 задания

YarRus12 committed:
Подготовлены заготовки для 2-3 задания

YarRus12 committed:
Сделаны заготовки для первого блока заданий

YarRus12 committed:
Выполнены задание 1-3

YarRus12 committed:
Выполнены задание 1-4

YarRus12 committed:
Выполнено задание 6, есть понимание как выполнить задание 5

YarRus12 committed:
Есть продвижения в 5 задании

YarRus12 committed:
Задание 5 выполнено

YarRus12 committed:
Выполнено задание 7

YarRus12 committed:
Выполнено задание 1

YarRus12 committed:
Выполнено задание 2

YarRus12 committed:
Выполнено задание 3

YarRus12 committed:
Выполнено задание 4

YarRus12 committed:
Выполнено задание 4

YarRus12 committed:
Выполнено задание 5

YarRus12 committed:
Выполнено задание 6

YarRus12 committed:
Подготовлена документация ко всем заданиям.
Осталось выполнить последнее задание

YarRus12 committed:
Все задания выполнены

YarRus12 committed:
Все задания выполнены и готовы к сдаче

DmitryChitalov commented on Урок 2. Практическое задание/task_1.py:
```diff
@@ -27,3 +27,40 @@
 """
+
+
+def arithmetics():
```
arithmetics
это имя удачное для массива, не для ф-ции

DmitryChitalov commented on Урок 2. Практическое задание/task_2.py:
```diff
@@ -17,3 +17,25 @@
+        print(f'Количество четных цифр: {even_digits}, Количество нечетных цифр: {odd_digits}')
+
+
+check_numbers(int(input('Введите число: ')))
```
можно упростить код
как в примере

DmitryChitalov commented on Урок 2. Практическое задание/task_3.py:
```diff
@@ -18,3 +18,19 @@
+
+
+# change_numbers(1234567890)
+change_numbers(int(input('Введите число: ')))
```
Введите число: 1230
Перевернутое число: 321

отрабатывает невернр

DmitryChitalov commented on Урок 2. Практическое задание/task_4.py:
```diff
@@ -10,3 +10,19 @@
+        print(f'Количество элементов - {iter_number}, их сумма - {result}')
+
+
+count(int(input('Сколько чисел будет в последовательности? ')), list(map(float, input('Введите числа: ').split())))
```
мы вводим только n
а числа создаются по формуле
"1 -0.5 0.25 -0.125 ..."

DmitryChitalov commented on Урок 2. Практическое задание/task_5.py:
```diff
@@ -19,3 +19,18 @@
+        ans_out(start_ord + 1, end_ord, [], 1)
+
+
+ans_out()
```
выполнено

DmitryChitalov commented on Урок 2. Практическое задание/task_6.py:
```diff
@@ -7,3 +7,29 @@
+        print(f'Жаль, но ты не угадал. Загаданное число - {number}')
+
+
+random_game()
```
выполнено

DmitryChitalov commented on Урок 2. Практическое задание/task_7.py:
```diff
@@ -12,4 +12,22 @@
+        else:
+            print('Равенство не выполняется')
+
+check_number(10)
```
задание не прочитано
1) Правой части выражения в рекурсивной ф-ции быть не должно!
2) n должно идти в сторону уменьшения. посмотрите пример

YarRus12 committed:
Задание 1 - изменено название функции на более удачное
Задание 2 - вывод функции выполнен как в примере
Задание 3 - исправлена ошибка с 0, отказался от пребразования в int
Задание 4 - обработка числа 1 в рекурсии исправлено
Задание 7 - обработка вводимого числа изменена. С каждой рекурсией оно движется в сторону уменьшения формула к сравнению вынесена запределы вызова функции.

YarRus12 committed:
Задание 1 - изменено название функции на более удачное
Задание 2 - вывод функции выполнен как в примере
Задание 3 - исправлена ошибка с 0, отказался от преобразования в int
Задание 4 - обработка числа 1 в рекурсии исправлено
Задание 7 - обработка вводимого числа изменена. С каждой рекурсией оно движется в сторону уменьшения формула к сравнению вынесена за пределы вызова функции.
